controllers/fenceagentsremediation_controller.go

```diff
@@ -79,10 +84,19 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		r.Log.Error(err, "failed to get FAR CR")
 		return emptyResult, err
 	}
-	// TODO: Validate FAR CR name to nodeName. Run isNodeNameValid
+	// Validate FAR CR name to match a nodeName from the cluster
+	r.Log.Info("Check FAR CR's name")
+	valid, err := utils.IsNodeNameValid(r.Client, req.Name)
+	if err != nil {
+		return emptyResult, err
+	}
+	if !valid {
+		r.Log.Info("consider recreating the CR - invalid CR's name to the cluster node name", "FAR CR's Name", req.Name)
```

Comment:
nit: I would skip the recommendation on how to fix the issue, you don't know why the issue happened. Also, the error message isn't clear to me... maybe "didn't find node matching CR's name" ? :thinking:

Reply:
Gotcha, and I agree 👍🏻